app/controllers/edit_directory_columns_controller.rb

```diff
@@ -26,16 +24,40 @@ def update
       raise Discourse::InvalidParameters, "Must have at least one column enabled"
     end
 
+    changes = ""
+    new_values = ""
+    previous_values = ""
+    staff_action_logger = StaffActionLogger.new(current_user)
+
     directory_column_params[:directory_columns].values.each do |column_data|
       existing_column = directory_columns.detect { |c| c.id == column_data[:id].to_i }
       if (
-           existing_column.enabled != column_data[:enabled] ||
+           existing_column.enabled != ActiveModel::Type::Boolean.new.cast(column_data[:enabled]) ||
              existing_column.position != column_data[:position].to_i
          )
-        existing_column.update(enabled: column_data[:enabled], position: column_data[:position])
+        change, new_value, previous_value =
+          staff_action_logger.edit_directory_columns_details(column_data, existing_column)
+
+        changes += change
+        new_values += new_value
+        previous_values += previous_value
+
+        existing_column.update(
+          enabled: column_data[:enabled],
+          position: column_data[:position].to_i,
+        )
       end
     end
 
+    details =
+      if changes.empty?
+        { Detail: "Nothing was changed" }
+      else
+        { Changes: changes }
```

Comment:
I pulled this PR locally, and here's what I see: 

```
New: {"name":"topic_count","enabled":"true"}{"name":"post_count","enabled":"true"}
Previous: {"name":"topic_count","enabled":"false"}{"name":"post_count","enabled":"false"}
Changes:
topic_count - enabled: true -> false
post_count - enabled: true -> false
```

Do we need the extra "Changes" section? Other log entries only have New/Previous, I think it would make sense to keep just New/Previous here as well.